src/subdomains/generic/kyc/services/kyc.service.ts

```diff
@@ -545,6 +596,39 @@ export class KycService {
     return this.updateKycStepAndLog(kycStep, user, data, ReviewStatus.MANUAL_REVIEW);
   }
 
+  async updateRecommendationData(kycHash: string, stepId: number, data: KycRecommendationData) {
+    const user = await this.getUser(kycHash);
+    const kycStep = user.getPendingStepOrThrow(stepId);
+
+    if (Config.formats.recommendationCode.test(data.key)) {
+      const recommendation = await this.recommendationService.getAndCheckRecommendationByCode(data.key);
+
+      await this.recommendationService.updateRecommendationInternal(recommendation, {
+        isConfirmed: true,
+        recommended: user,
+        type: RecommendationType.RECOMMENDATION_CODE,
+        kycStep,
+        confirmationDate: new Date(),
+      });
+    } else {
+      // create new recommendation
+      await this.recommendationService.createRecommendationByRecommended(
+        Config.formats.ref.test(data.key) ? RecommendationType.REF_CODE : RecommendationType.MAIL,
+        data.recommendedAlias,
+        user,
+        kycStep,
+        Config.formats.ref.test(data.key) ? { refCode: data.key } : { mail: data.key },
+      );
```

Comment:
Maybe do the data.key check all in recommendation service? Currently we do things twice.